packages/apollo-engine-reporting/src/agent.ts

```diff
@@ -329,8 +372,27 @@ export class EngineReportingAgent<TContext = any> {
     handleLegacyOptions(this.options);
   }
 
+  public executableSchemaIdGenerator(schema: string | GraphQLSchema) {
+    if (this.lastSeenExecutableSchemaToId?.executabeSchema === schema) {
+      return this.lastSeenExecutableSchemaToId.executableSchemaId
+    }
+    const id = computeExecutableSchemaId(schema);
+
+    this.lastSeenExecutableSchemaToId = {
+      executabeSchema: schema,
+      executableSchemaId: id,
+    }
```

Comment:
Is this object map for _the future_?  It seems it's only definitively/wholesale overwritten right now rather than having properties assigned into it.  If so, a comment would seem warranted.